src/js/utils.js

```diff
@@ -445,6 +445,46 @@ function isThirdPartyDomain(domain1, domain2) {
   return false;
 }
 
+/**
+ * checks to see if a given url has firstparty protections scripts run on it
+ * @param {String} domain
+ * @param {Object} firstPartiesList a second fqdn
+ *
+ * @return {Boolean} true if the domains are third party
+ */
+function firstPartyProtectionsEnabled(domain, badger) {
+  // if the firstparties list hasn't been set yet, fetch it and set it
+  if (!badger.firstPartiesList) {
+    let manifestJson = chrome.runtime.getManifest();
+    let firstParties = [];
+
+    for (let contentScriptObj of manifestJson.content_scripts) {
+      // only include parts from content scripts that have firstparties entries
+      if (contentScriptObj.js[0].includes("/firstparties/")) {
+        let extractedUrls = [];
+        for (let match of contentScriptObj.matches) {
+          extractedUrls.push(window.extractHostFromURL(match));
+        }
+        firstParties.push(extractedUrls);
+      }
+    }
+    // clean up the gathered sets of url matches into a single flat list
+    badger.firstPartiesList = [].concat.apply([], firstParties);
+  }
+  // check for presence of given domain in firstparties list
+  for (let url_pattern of badger.firstPartiesList) {
+    // account for wildcards in entries on firstPartiesList & avoid false positives
```

Comment:
Nit, but: "avoid false positives" is something you'd write in the body of your commit message that improves some check to avoid false positives. It doesn't make sense as a comment though, I don't think, as it refers to something that's irrelevant at this point.